app_dart/test/service/luci_build_service_test.dart

```diff
@@ -0,0 +1,316 @@
+            Response(
+                searchBuilds: SearchBuildsResponse(builds: <Build>[
+              Build(
+                id: 998,
+                builderId: BuilderId(
+                  project: 'flutter',
+                  bucket: 'prod',
+                  builder: 'Linux',
+                ),
+                status: Status.started,
+              )
+            ]))
+          ],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      await service.cancelBuilds('cocoon', 1, 'abc', 'new builds');
+      verify(mockBuildBucketClient.batch(any)).called(2);
+    });
+    test('unsupportedRepo', () async {
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      expect(
+          () async => await service.cancelBuilds(
+                'notsupported',
+                1,
+                'abc',
+                'new builds',
+              ),
+          throwsA(const TypeMatcher<BadRequestException>()));
+    });
+  });
+
+  group('failedBuilds', () {
+    setUp(() {
+      serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+      config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+      mockBuildBucketClient = MockBuildBucketClient();
+    });
+    test('emptyList', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final List<Build> result = await service.failedBuilds('cocoon', 1, 'abc');
+      expect(result, isEmpty);
+    });
+    test('withFailures', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+                searchBuilds: SearchBuildsResponse(builds: <Build>[
+              Build(
+                id: 998,
+                builderId: BuilderId(
+                  project: 'flutter',
+                  bucket: 'prod',
+                  builder: 'Linux',
+                ),
+                status: Status.failure,
+              )
+            ]))
+          ],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final List<Build> result = await service.failedBuilds('cocoon', 1, 'abc');
+      expect(result, hasLength(1));
+    });
+  });
+  group('rescheduleBuild', () {
+    setUp(() {
+      serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+      config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+      mockBuildBucketClient = MockBuildBucketClient();
+    });
+    test('worksProperly', () async {
```

Comment:
Nit: the dart convention is to use a short sentence as test name rather than identifier-ish.

Reply:
Done